iroh/src/client.rs

```diff
@@ -433,6 +434,51 @@ where
         self.rpc.rpc(BlobDeleteBlobRequest { hash }).await??;
         Ok(())
     }
+
+    /// Share a blob.
+    pub async fn share(
+        &self,
+        hash: Hash,
+        blob_format: BlobFormat,
+        no_derp: bool,
+        derp_only: bool,
+    ) -> Result<BlobTicket> {
+        let NodeStatusResponse { addr, .. } = self.rpc.rpc(NodeStatusRequest).await??;
+        let node_addr = if no_derp {
+            NodeAddr::new(addr.node_id).with_direct_addresses(addr.direct_addresses().copied())
+        } else if derp_only {
+            if let Some(url) = addr.derp_url() {
+                NodeAddr::new(addr.node_id).with_derp_url(url.clone())
+            } else {
+                addr
+            }
+        } else {
+            addr
+        };
+
+        let ticket = BlobTicket::new(node_addr, hash, blob_format).expect("correct ticket");
+
+        Ok(ticket)
+    }
+
+    /// Get the status of a blob.
+    pub async fn status(&self, hash: Hash) -> Result<BlobStatus> {
+        // TODO: this could be implemented more efficiently
+        let reader = self.read(hash).await?;
+        Ok(BlobStatus {
+            size: reader.size,
+            is_complete: reader.is_complete,
+        })
+    }
+}
+
+/// Status information about a blob.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct BlobStatus {
```

Comment:
Would also prefer an enum here. E.g. "partial" is a pretty wide range from 1/1TiB to 999GiB/1TiB. We might want to add some more info here.

Reply:
but that information is not known, the store doesn't expose this easily

Reply:
Yes, I know. And it is not free to compute. What I'm saying is that with an enum it is easier to evolve it as more information becomes available.

Reply:
done